src/ol/layer/Layer.js

```diff
@@ -450,7 +450,7 @@ class Layer extends BaseLayer {
       this.mapPrecomposeKey_ = listen(
         map,
         RenderEventType.PRECOMPOSE,
-        function (evt) {
+        (evt) => {
```

Comment:
With an arrow function, you could use `addEventListener()` instead of `listen()` and get rid of the `mapPrecomposeKey_`.  Then instead you would store the `mapPrecomposeFunction_` for later use in `removeEventListener()` instead of `unlisten()`.